### PR TITLE
Update build doc

### PR DESCRIPTION
**Pull request by lvhan028**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Update user documents to enhance readability and usability

## Modification

- [x] docs/en/build/linux.md
- [x] docs/zh_cn/build/linux.md
- [x] docs/en/build/windows.md
- [x] docs/zh_cn/build/windows.md
- [x] docs/en/build.md
- [x] docs/en/index.rst
- [x] docs/zh_cn/build.md
- [x] docs/zh_cn/index.rst

### CODE REVIEW

**Comment by triple-Mu** [comment]
pytorch1.11.0的torch.onnx.export接口传参有变动 strip_doc_string参数貌似不需要了

**Comment by lvhan028** [comment]
> pytorch1.11.0的torch.onnx.export接口传参有变动 strip_doc_string参数貌似不需要了 还有官网的文档 2615c41d97310fd9baa927445249f48 例子路径缺少mmdet

It will be specified in `get_started.md`. Another PR will address this issue.

**Comment by triple-Mu** [comment]
@lvhan028  感谢您的回复！我会继续测试win10下的mmdeploy

File: docs/zh_cn/build/linux.md [hunk 1]
```diff
@@ -1 +1,416 @@
...
+    <td>设置目标设备。当有多个设备时，设备名称之间使用分号隔开。 比如，-DMMDEPLOY_TARGET_DEVICES="cpu;cuda"</td>
+  </tr>
+  <tr>
+    <td>MMDEPLOY_TARGET_BACKENDS</td>
+    <td>{"trt", "ort", "pplnn", "ncnn", "openvino"}</td>
+    <td>N/A</td>
+    <td> <b>默认情况下，SDK不设置任何后端</b>, 因为它与应用场景高度相关。 当选择多个后端时， 中间使用分号隔开。比如，<pre><code>-DMMDEPLOY_TARGET_BACKENDS="trt;ort;pplnn;ncnn;openvino"</code></pre>
+    构建时，几乎每个后端，都需传入一些路径变量，用来查找依赖包。<br>
+    1. <b>trt</b>: 表示 TensorRT。需要设置 <code>TENSORRT_DIR</code> 和 <code>CUDNN_DIR</code>。
+<pre><code>-DTENSORRT_DIR=${TENSORRT_DIR}<br>-DCUDNN_DIR=${CUDNN_DIR}</code></pre>
```
Comment (irexyc) [inline docs/zh_cn/build/linux.md:303]: Add space between the -D command would be better.

File: docs/zh_cn/build/linux.md [hunk 2]
```diff
@@ -1 +1,416 @@
...
+
+    ```bash
+    sudo apt-get install -y libssl-dev
+    wget https://github.com/Kitware/CMake/releases/download/v3.20.0/cmake-3.20.0.tar.gz
+    tar -zxvf cmake-3.20.0.tar.gz
+    cd cmake-3.20.0
+    ./bootstrap
+    make
+    sudo make install
+    ```
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply docs/zh_cn/build/linux.md:40]: > There is no need to build cmake.

The default cmake on Ubuntu 16.04 is less than 3.14. That's the reason we provide an example of building cmake

File: docs/en/build/linux.md [hunk 3]
```diff
@@ -1 +1,427 @@
...
+      -DMMDEPLOY_BUILD_SDK=ON \
+      -DMMDEPLOY_BUILD_SDK_PYTHON_API=ON \
+      -DMMDEPLOY_TARGET_DEVICES="cuda;cpu" \
+      -DMMDEPLOY_TARGET_BACKENDS=trt \
+      -DMMDEPLOY_CODEBASES=all \
+      -Dpplcv_DIR=${PPLCV_DIR}/cuda-build/install/lib/cmake/ppl \
+      -DTENSORRT_DIR=${TENSORRT_DIR} \
+      -DCUDNN_DIR=${CUDNN_DIR}
+
+  make -j$(nproc) && make install
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply docs/en/build/linux.md:417]: > what is the different?
> 
> ```shell
> cmake --build . -- -j$(nproc) && cmake --install .
> ```

In cmake 3.14, `cmake --install` is not supported.

[the remark this answers is not in the document]
Comment (lvhan028) [reply docs/en/build/linux.md:417]: > what is the different?
> 
> ```shell
> cmake --build . -- -j$(nproc) && cmake --install .
> ```

CMake v3.14 doesn't support `cmake --install`

**Review by triple-Mu — APPROVED** [review]
LGTM

File: docs/en/build/windows.md [hunk 4]
```diff
@@ -1 +1,374 @@
...
+  ```PowerShell
+  cd $env:MMDEPLOY_DIR
+  mkdir build -ErrorAction SilentlyContinue
+  cd build
+  cmake .. -G "Visual Studio 16 2019" -A x64 -T v142 `
+    -DMMDEPLOY_BUILD_SDK=ON `
+    -DMMDEPLOY_TARGET_DEVICES="cuda" `
+    -DMMDEPLOY_TARGET_BACKENDS="trt" `
+    -DMMDEPLOY_CODEBASES="all" `
+    -Dpplcv_DIR="$env:PPLCV_DIR/pplcv-build/install/lib/cmake/ppl" `
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply docs/en/build/windows.md:351]: All environment path variables,  such as $env:TENSORRT_DIR, $env:CUDNN_DIR and $env:pplcv_DIR, have to be enclosed by quotation marks when passed to cmake. I've correct them.

File: docs/en/build.md [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
 
-We provide building methods for both physical and virtual machines. For virtual machine building methods, please refer to
-[how to use docker](tutorials/how_to_use_docker.md). For physical machine, please follow the steps below.
+## Download MMDeploy
 
-### Preparation
 
-- Download MMDeploy
+  ```bash
+  git clone -b master git@github.com:open-mmlab/mmdeploy.git MMDeploy
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply docs/en/build.md:7]: We are not going to suggest any proxy tools.

File: docs/zh_cn/build/linux.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (irexyc) [reply docs/zh_cn/build/linux.md:40]: What about download the binary version like 
https://github.com/Kitware/CMake/releases/download/v3.20.1/cmake-3.20.1-linux-x86_64.tar.gz

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

**Review by irexyc — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**